### PR TITLE
Refactor/ra 137 refactor construction preprocess

### PR DESCRIPTION
**Pull request by Gennadii-T**

### CODE REVIEW

File: api/src/main/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImpl.java [hunk 1]
```diff
@@ -203,23 +203,16 @@ public Integer calculateTxSize(NetworkIdentifierType networkIdentifierType,
     UnsignedTransaction unsignedTransaction;
     try {
       unsignedTransaction = createUnsignedTransaction(networkIdentifierType,
-          operations, ttl, !ObjectUtils.isEmpty(depositParameters) ? depositParameters
-              : new DepositParameters(Constants.DEFAULT_KEY_DEPOSIT.toString(),
+          operations, ttl, depositParameters != null ?
```
Comment (Kammerlo) [inline api/src/main/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImpl.java:206]: DepositParameters could be an empty object

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java [hunk 2]
```diff
@@ -94,6 +98,26 @@ void outputMoreThanInputTest() {
     assertEquals("The transaction you are trying to build has more outputs than inputs", exception.getError().getMessage());
   }
 
+  @Test
+  void calculateTxSize_whenUnsignedTransactionAddressesInvalid_thenThrowException() {
```
Comment (BerezinD) [inline api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:102]: Do we have only one unit test for the `calculateTxSize()` method?

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java [hunk 3]
```diff
@@ -94,6 +98,26 @@ void outputMoreThanInputTest() {
...
+    List<Operation> operations = Collections.singletonList(givenOperation());
+    try (MockedStatic<CardanoAddressUtils> mockedAddressUtil = Mockito.mockStatic(
+        CardanoAddressUtils.class)) {
+      mockedAddressUtil.when(() -> CardanoAddressUtils.getEraAddressType(anyString()))
+          .thenReturn(null);
+      mockedAddressUtil.when(() -> CardanoAddressUtils.isEd25519KeyHash(anyString()))
```
Comment (BerezinD) [inline api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:108]: This mock is not reachable

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java [hunk 4]
```diff
@@ -94,6 +98,26 @@ void outputMoreThanInputTest() {
...
+          .thenReturn(false);
+
+      ApiException result = assertThrows(ApiException.class,
+          () -> cardanoService.calculateTxSize(NetworkIdentifierType.CARDANO_TESTNET_NETWORK,
+              operations, 0, null));
+
+      assertEquals(RosettaErrorType.INVALID_ADDRESS.getMessage(), result.getError().getMessage());
+      assertEquals(RosettaErrorType.INVALID_ADDRESS.getCode(), result.getError().getCode());
+      assertTrue(result.getError().isRetriable());
```
Comment (BerezinD) [inline api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:117]: It's a good practice to check service interactions to make the flow explicit

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java (hunk 3 above)
Comment (Gennadii-T) [reply api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:108]: Why? This mock is needed for the #extractSignaturesFromAddress method
image

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java (hunk 2 above)
Comment (Gennadii-T) [reply api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:102]: Yeah, our services unit test coverage is rather poor

File: api/src/main/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImpl.java (hunk 1 above)
Comment (Gennadii-T) [reply api/src/main/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImpl.java]: Reverted

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java (hunk 4 above)
Comment (Gennadii-T) [reply api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:118]: done

File: api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java (hunk 3 above)
Comment (Gennadii-T) [reply api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java:109]: Added verify() check

**Review by Kammerlo — APPROVED** [review]

**Pull request merged**